src/app/content/highlights/reducer.ts

```diff
@@ -74,26 +77,22 @@ const reducer: Reducer<State, AnyAction> = (state = initialState, action) => {
     case getType(actions.updateHighlight): {
       if (!state.highlights) { return state; }
 
-      const oldHiglightIndex = state.highlights.findIndex(
-        (highlight) => highlight.id === action.payload.id);
-      if (oldHiglightIndex < 0) { return state; }
+      const currentHighlight = getHighlightByIdFromHighlights(state.highlights, action.payload.id)
+        || getHighlightByIdFromSummaryHighlights(state.summary.highlights, action.payload.id);
```

Comment:
i'm not sure this logic is working correctly. i think you should process the current page highlights and the summary highlights separately, and each one should be in a function that correctly noops and returns the input state if no action is required. 

`updateSummaryHighlightsDependOnFilters` it has a noop if the chapter doesn't match filters, but doesn't have one for changing a color that did not match filters to another color that also doesn't match filters.


edit: the `updateHighlights` function does this already, it returns the input state if no action is necessary, so there is no reason to separately check to see if action is necessary on its behalf.

Reply:
I'll add test for this case in #424 pr when this pr become approved